src/widget/tooltipqopengl.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <QPoint>
+#include <QToolTip>
+
+class QTimer;
+class WGLWidget;
+
+// Tooltips don't work for the qopengl-based WGLWidget. This
+// singleton mimics the standard tooltip behaviour for them.
+class ToolTipQOpenGL : public QObject {
+    Q_OBJECT
+
+    bool m_active{true};
+    std::unique_ptr<QTimer> m_timer;
```

Comment:
To be able to use forward declaration.

But since tooltipqopengl.h is only included in four cpp files I agree that the simplicity of using QTimer as a member is preferable. I'll make the change.

Reply:
DONE (removed unique_ptr, using QTimer as member)

Reply:
May I ask why you would prefer forward declaration here? I assume to speed up compile-times?

Reply:
Yes, compile time and also to keep dependencies between modules to a minimum. If you include headers from other headers when it's not strictly necessary, before you know it touching something in one header can trigger recompilation of many files that are not even remotely related.  In this case, this header is only included from a couple of cpp files, so there is no such risk of excessively propagating dependencies and the use of the forward declaration was exaggerated (also because we are going to touch QTimer).

It also helps to structure the code in modules without having to drag along a lot of public include paths of other modules. The mixxx CMakeLists.txt is really bad practice, one huge CMakeLists.txt instead of having proper modules in folders. But improving that will be a huge undertaking...

Reply:
I fully I agree. In the case of Qt headers, I would argue it doesn't really hurt to include the headers when possible since those headers are fairly stable.